app/controllers/users_controller.rb

```diff
@@ -11,7 +11,9 @@ def create
       redirect_to '/profile'
     else
       flash[:notice] = @user.errors.full_messages.uniq.to_sentence
-      @user_params = user_params
+      if flash[:notice].include?("Email has already been taken")
+        @user.update(email: nil)
+      end
```

Comment:
Since form_for is specified to be for @user, it made the user_params irrelevant to the form but @user was completely relevant